src/main/java/duke/Duke.java

```diff
@@ -0,0 +1,270 @@
+    }
+
+    // add a new deadline
+    private static void addDeadline(String taskInfo) {
+        Task newTask = createDeadline(taskInfo);
+        tasks.add(newTask);
+        System.out.print(breakLine()
+                + "\tadded:\n\t\t" + newTask + '\n'
+                + "\t(total: " + tasks.size() + " tasks)\n"
+                + breakLine());
+    }
+
+    // create new event
+    private static Event createEvent(String taskInfo) {
+        String description, from, to;
+        String[] info = taskInfo.split("#from", 2);
+        description = info[0];
+        info = info[1].split("#to", 2);
+        from = info[0];
+        to = info[1];
+        return new Event(description, from, to);
+    }
+
+    // add a new event
+    private static void addEvent(String taskInfo) {
+        Task newTask = createEvent(taskInfo);
+        tasks.add(newTask);
+        System.out.print(breakLine()
+                + "\tadded:\n\t\t" + newTask + '\n'
+                + "\t(total: " + tasks.size() + " tasks)\n"
+                + breakLine());
+    }
+
+    // list all tasks
+    private static void listTask() {
+        System.out.println(breakLine() + "\tThese are the tasks you have (" + tasks.size() + " tasks):");
+        int order = 1;
+        for (Task task : tasks) {
+            System.out.print("\t" + order + ". " + task.toString() + '\n');
+            order++;
+        }
+        System.out.print(breakLine());
+    }
+
+    // mark the task
+    private static void markTask(String taskNumber) {
+        int taskIdx = Integer.parseInt(taskNumber) - 1;
+        try {
+            Task currentTask = tasks.get(taskIdx);
+            currentTask.mark();
+            System.out.print(breakLine()
+                    + "\tNice! I've marked this task as done :D\n\t\t"
+                    + currentTask + '\n'
+                    + breakLine());
+        } catch (IndexOutOfBoundsException e) {
+            System.out.print(breakLine()
+                    + "\t(!) Invalid task number. Please try again :(\n"
+                    + breakLine());
+        }
+    }
+
+    // unmark the task
+    private static void unmarkTask(String taskNumber) {
+        int taskIdx = Integer.parseInt(taskNumber) - 1;
+        try {
+            Task currentTask = tasks.get(taskIdx);
+            currentTask.unmark();
+            System.out.print(breakLine()
+                    + "\tOh. I've unmarked this task as not done yet :(\n\t"
+                    + currentTask + '\n'
+                    + breakLine());
+        } catch (IndexOutOfBoundsException e) {
+            System.out.print(breakLine()
+                    + "\t(!) Invalid task number. Please try again :(\n"
+                    + breakLine());
+        }
+    }
+
+    // delete Task
+    private static void deleteTask(String taskNumber) {
+        int taskIdx = Integer.parseInt(taskNumber) - 1;
+        try {
+            Task currentTask = tasks.get(taskIdx);
```

Comment:
Would be good to spell out the full form (eg. taskIndex rather than taskIdx) for better readability